### PR TITLE
fix the JNI CF handle leak

### PR DESCRIPTION
**Pull request by gwnet**
I got the memory leak now. It is 100% precise. 

Actually `dropColumnFamily` method will not free `ColumnFamily` at all. In C++ world, RocksDB provide one method below.

```C++
Status DB::DestroyColumnFamilyHandle(ColumnFamilyHandle* column_family) {
  delete column_family;
  return Status::OK();
}
```

But in java world JNI did not expose it at all!~~~

### CODE REVIEW

File: java/rocksjni/rocksjni.cc [hunk 1]
```diff
@@ -1707,6 +1707,23 @@ void Java_org_rocksdb_RocksDB_dropColumnFamily(JNIEnv* env, jobject /*jdb*/,
...
 }
 
+/*
+ * Class:     org_rocksdb_RocksDB
+ * Method:    destroyColumnFamilyHandle
+ * Signature: (JJ)V;
+ */
+void Java_org_rocksdb_RocksDB_destroyColumnFamilyHandle(JNIEnv* env,
+                                                        jobject /*jdb*/,
+                                               jlong jdb_handle,
```
Comment (riversand963) [inline java/rocksjni/rocksjni.cc]: Nit: indentation.

Comment (gwnet) [reply java/rocksjni/rocksjni.cc]: I need two reviewer approve to merge it?

**Comment by gwnet** [comment]
call for another approver for memory leak risk

**Comment by gwnet** [comment]
> 1. As it currently stands, the javadoc for `RocksDB#dropColumnFamily(ColumnFamilyHandle)` does not match

Hello Buddy, after review Java code carefully, I believe you are absolutely correct. I will close this PR. 
Could you please help me review another my pull request that is for JNI list and get performance with custom Java comparator. see below link:

**Comment by gwnet** [comment]
> So I am trying to understand the problem that this solves. A few observations,
> 
> 1. As it currently stands, the javadoc for `RocksDB#dropColumnFamily(ColumnFamilyHandle)` does not match the comments in the C++ header file `for virtual Status DB::DropColumnFamily(ColumnFamilyHandle* column_family)`. The C++ comments, make it clear that that function is for dropping (i.e. deleting) a column family, not for closing it.
> 2. `org.rocksdb.ColumnFamilyHandle` implements `java.lang.AutoCloseable`. When you call `ColumnFamilyHandle#close()` it calls into JNI and deletes the underlying C++ object for the column family handle. Doing so, should prevent any memory leak.
> 
> The Java API intentionally does not expose `DB::DestroyColumnFamilyHandle` because it has no need to, you can instead simply call `ColumnFamilyHandle#close()`.
> 
> Unless I have missed anything? I don't think this change is needed as the functionality to close (delete the underlying memory) for a column family already exists for you in the Java API.

Hello Buddy, after review Java code carefully, I believe you are absolutely correct. I will close this PR.
Could you please help me review another my pull request that is for JNI list and get performance with custom Java comparator. the pull id is 4425
_https://github.com/facebook/rocksdb/pull/4425

**Pull request closed without merging**